java/Anton-Kozyk/src/com/zoolatech/lecture4/tasks/_4/WeightUnit.java

```diff
@@ -0,0 +1,40 @@
+package com.zoolatech.lecture4.tasks._4;
+
+public enum WeightUnit {
+    GRAM("g", 1),
+    KILOGRAM("kg", 1000),
+    OUNCE("oz", 28.35),
+    POUND("lb", 453.6),
+    TON("t", 1000000);
+
+    private final String abbreviation;
+    private final double inGram;
+
+    static WeightUnit getValueByName(String name) {
+        try {
+            String unitName = name.toUpperCase();
+            return valueOf(unitName);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getLocalizedMessage());
+            return null;
+        }
+    }
+
+    WeightUnit(String abbreviation, double inGram) {
+        this.abbreviation = abbreviation;
+        this.inGram = inGram;
+    }
+
+    String getAbbreviation() {
+        return abbreviation;
+    }
+
+    double convert(int value, String sourceUnit) {
+        WeightUnit unit = getValueByName(sourceUnit);
```

Comment:
prefer to define methods in an order they are used: if some method (`convert`) uses other method (`getValueByName`), other method should be defined after the first method - this would allow to read your code in the order